databuilder/query_language.py

```diff
@@ -274,9 +277,11 @@ class FloatPatientSeries(NumericFunctions, PatientSeries):
 
 def parse_date_if_str(value):
     if isinstance(value, str):
-        return datetime.date.fromisoformat(value)
-    else:
-        return value
+        try:
+            value = datetime.date.fromisoformat(value)
+        except ValueError:
+            pass
```

Comment:
This is another consequence of applying the parsing in `BaseSeries.is_in()`. We want to raise errors for malformed dates but we can only do that if know that the string is supposed to be a date, which means only calling this function in date contexts.